my-app/src/AccountDetails.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import PaymentHistoryTable from './PaymentHistoryTable';
+
+interface AccountDetailsProps {
+  id: string;
+}
+
+interface AccountInfo {
+  name: string;
+  socialSecurityNumber: string;
+  dateOfBirth: string;
+  address: string;
+  creditScore: number;
+  email: string;
+  phoneNumber: string;
+  cardNumber: string;
+}
+
+interface AccountInfo {
+  id: string;
+  name: string;
+  socialSecurityNumber: string;
+  dateOfBirth: string;
+  address: string;
+  creditScore: number;
+  email: string;
+  phoneNumber: string;
+  cardNumber: string;
+}
+
+interface PaymentRecord {
+  date: string;
+  description: string;
+  amount: number;
+  balance: number;
+}
+
+interface AccountDetails {
+  accountInfo: AccountInfo;
+  paymentHistory: PaymentRecord[];
+}
+
+
+
+const AccountDetails: React.FC = () => {
+  const { id } = useParams<AccountDetailsProps>();
+  const [accountInfo, setAccountInfo] = useState<AccountInfo>(null);
+
+  useEffect(() => {
+    fetchAccountDetails(id)
+  }, [id]);
+
+
+  const fetchAccountDetails = async (accountId: string): Promise<void> => {
+    try {
+      const response = await fetch(`https://api.apg.com/accounts/${accountId}`);
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+
+      const accountDetails: AccountDetails = await response.json();
+      setAccountInfo(accountDetails);
+    } catch (error) {
+      console.error('Error fetching account details:', error);
+      throw error;
```

Comment:
Can you please handle the error gracefully here? Talk to the designer about what the desired behavior is if we don't get the account details successfully